### ISSUE TITLE
What are the must haves, should haves, could haves, won't have for the features of the blog?

### ISSUE THREAD

**Issue by mark-andrews** [ntupsychology/ntur_blog#1]
### Must haves    

Here is the list of features that either I personally won't compromise on or else we as a group think *must* all be present for the blog to be viable.   Anything marked with :heavy_check_mark: is already done, and anything marked with :heavy_multiplication_x: is not done.

- All posts are written in markdown, which allows e.g hyperlinks, styling and formatting (e.g. italics, bold, bullet lists, numbered lists), including images and so on :heavy_check_mark:.
- All blog config files written in config languages like toml, yaml :heavy_check_mark:.
- All markdown and config files under version in Git, with all users pushing/pulling to a remote Git repository to add and edit content and blog settings:heavy_check_mark:.
- LaTeX math written in the markdown and is properly rendered with [mathjax](https://www.mathjax.org/):heavy_check_mark:.
- All code blocks in the markdown are syntax highlighted :heavy_check_mark:.
- Hosted at a custom domain name :heavy_check_mark:.
- Tags for blog topics, with each tag leading to a page of all and only posts on that topic :heavy_multiplication_x:.
- Author of each post shown after title, with allowance for multiple authors per post :heavy_multiplication_x:.

Here's how we are with these: 

- The **content in markdown** and configs in toml/yaml are inherent features of Hugo. 
- **Version control of all content and config files** and adding, editing, deleting and reconfiguring the site using Git operations are inherent features of using the GitHub + netlify approach to hosting.  
- I added **LaTeX maths** rendering and code syntax highlighting as of version https://github.com/ntupsychology/ntur_blog/commit/0bd742baccb1145ebb7b739cd04ef169aa62853c. 
- As of https://github.com/ntupsychology/ntur_blog/commit/c8b00e1b1dea8dae4464b46ec1cee561ad3d15c8 we've had a working **custom domain name**, i.e. http://ntur.ntupsychology.net. 
- As of the latest version, ie., https://github.com/ntupsychology/ntur_blog/commit/54b2040ee9521c69641d45cdfefc70327f95c6c3, there are no **topic tags**. However, unless I am gravely mistaken topic tags are a widely used feature of Hugo based websites and blogs and so I'm anticipating that this won't be a problem.  
- Again, as of the latest version, ie., https://github.com/ntupsychology/ntur_blog/commit/54b2040ee9521c69641d45cdfefc70327f95c6c3, there are no **author names** after each post. However, currently, posts are all markdown files, and markdown files have yaml headers with e.g. the title, date and author(s). Using the current minimal theme, the author names are not rendered for each post. It is just the title and the date. However, I think that with some tinkering, this should be easy to fix. 

### Should haves 

- Posts to be written in Rmarkdown and not just markdown :heavy_multiplication_x:.
- Inline BibTeX citations and bibliography at the end of the blog post :heavy_multiplication_x:.
- Non blog pages, e.g. "about" pages  :heavy_multiplication_x:.

Here's prognosis with these items. 

- All Hugo sites use markdown for content. But more recently, [blogdown](https://bookdown.org/yihui/blogdown/) *seems* (I haven't looked or experimented hard enough to know with certainty) to give us the ability to just as easily **use RMarkdown** for content in Hugo sites. So I will look into this as a matter of priority. 
- Obviously, **citations and bibliographies** and an essential feature of academic writing. Markdown, when used with the pandoc processor makes using BibTeX for inline citations and bibliographies about as easy as one could possibly ask for. However, I am not use if this works with Hugo. I don't know if Hugo uses pandoc on its back-end and even if it does, or doesn't, I'm don't know if it allows for the use of a BibTeX processor. On the other hand, RMarkdown does allow for BibTeX citations, and [blogdown](https://bookdown.org/yihui/blogdown/)  uses RMarkdown for Hugo sites (unless I am misunderstanding something), and so maybe something's possible that way. 
- The **non blog post pages** seem easy enough to make. After we get up and running, and have a real post, I think putting an "about" page together should be a priority. 

### Could haves 

- Names of all authors on each post being a link leading to a page of all and only posts from that author :heavy_multiplication_x:.
- RSS feeds for all posts :heavy_multiplication_x:.
- Comments on posts :heavy_multiplication_x:.

Here's prognosis with these items. 

- I'm not sure exactly how to do the **author posts pages**. I'd guess it doable with some Hugo theme editing. But do we really need it? With the topic tags, it makes sense: Reader may want to read all the posts on a given topic, and so you don't want them hunting them down. But do we anticipate any real need for readers wanting to read all the posts by just one of us. I doubt it. 
- Personally, I just don't use **RSS** feeds, and they feel a bit Web 1.0 and unnecessary to me. Personally, I'd rather use social media (well Twitter really) to advertise and disseminate new posts and get readers. Still, RSS feeds may have real value and so this issue maybe should be discussed. In case it matters, it actually looks very straightforward to set up RSS feeds in Hugo. 
- The issue of **comments on posts** is a important one. On the one hand, comments and discussion on posts have the potential of turning a monologue into a rich and ongoing dialogue, thus adding to the quality of the content, and developing meaningful connections with readers. On the other hand, the *vast* majority of comments online in general are unreflective, unthinking, inane, petulant and so on, or they're simply spam. Filtering pure spam is one task, and is still error prone, but I'm more unconcerned about the low quality comments. Even if the commentators were not trolling, I don't think there is that much to be gained from some hastily written one-line comment by someone anonymous. Even when it is not trolling, these types of comments mostly just add noise. Much better would be properly and intelligent comments where the author is not anonymous. Another relevant point for this issue is that Hugo, as a static site generator, does not deal with access control and user accounts. That means that a commentary system would need to rely on a third party system like [disqus](https://disqus.com). Personally speaking, I really don't like the look of the disqus frame at the end of the page, and so I'm not enthusiastic about this solution. In summary, however, I think I may have a workable, if not ideal, solution to have we can have a commentary system without dealing with all the junk, and without rely on a third party system. I'll raise my suggestions in another Issue. 

### Won't haves

This list could go on and on. But here are a few:

- Ads of any kind  :heavy_multiplication_x:.
- Twitter feeds in a side column  :heavy_multiplication_x:.
- Anything cluttered and visually noisy  :heavy_multiplication_x:.
- Anything that remotely resembles bells and whistles  :heavy_multiplication_x:.

**Comment by Mark-Torrance** [comment]
Mark, massive thanks for doing this.

Should have: 

- Some sort of chronological index. I find it really annoying when I can't see this, in any blog, but as this is meant to document our journey into the light, chronology is important.
- Reader comments and discussion? I think this is potentially really useful (if moderated). Often as valuable as the blog itself. Hugo support Disqus, but I think this comes with adverts unless you pay. [Edit: I really must read properly before commenting (ironically). I agree with what you say here.]

Should not have:
- Requirement to use RMarkdown. Option yes, but requirement no. We need to make it as easy to post as possible. For many posts this won't be necessary.

I think I agree with the rest. Posts need to give authors, but I am not sure that we need an author index. We definitely need an "About".

**Comment by Mark-Torrance** [comment]
Oh, and I use RSS to monitor blogs, fwiw. And I'm guessing R aggerators like R-Bloggers do. On a quick Google I can't find any stats for how blogs are accessed, but even if 10% is via RSS, if it's easy to add it's worth doing?

**Comment by mark-andrews** [comment]
It case it wasn't clear, I never meant that posts should only be written in RMarkdown. I just meant that it would be nice to have the option of writing in RMarkdown (i.e. markdown with embedded R code chunks) too. 

I agree with having a chronological index of posts. I'd say that will be no problem to implement. Likewise, I think the RSS feature should be bumped from a "could have" to a "should have" feature, for reasons stated. 

For all the as yet unimplemented must haves, and for all the should haves, and maybe even all the could haves, I think it would be best for each to have their own Issue page. That way we can have "assignees" to solve them and when they're done we can do the psychologically satisfying "closing" of the Issue. For this purpose, I've added a few more Issue tracker labels.

**Comment by Mark-Torrance** [comment]
RMarkdown: Good. That makes perfect sense. Just to be clear (for other people looking at this) you can still have code chunks in ordinary markdown, but you don't get syntax highlighting (I think), and obviously you can't actually execute them.

I spent a while looking at themes a couple of weeks ago.  [Cactus](https://themes.gohugo.io/hugo-theme-cactus-plus/) has, I think, has just the right number of bells and whistles - Disqus, RSS, index, syntax highlighting, social media links - but is otherwise minimal. And of course you can take out things we don't want. It might save time to start from this?

[Oh, and "aggregators" in my previous post - now edited. I don't know where "exaggregators" came from. Nice word, though.]
